integration-cli/docker_cli_build_test.go

```diff
@@ -5327,9 +5329,9 @@ func (s *DockerSuite) TestBuildResourceConstraintsAreUsed(c *check.C) {
 	if err := json.Unmarshal([]byte(cfg), &c2); err != nil {
 		c.Fatal(err, cfg)
 	}
-	if c2.Memory == 67108864 || c2.MemorySwap == -1 || c2.CpusetCpus == "0" || c2.CpusetMems == "0" || c2.CpuShares == 100 || c2.CpuQuota == 8000 {
-		c.Fatalf("resource constraints leaked from build:\nMemory: %d, MemSwap: %d, CpusetCpus: %s, CpusetMems: %s, CpuShares: %d, CpuQuota: %d",
-			c2.Memory, c2.MemorySwap, c2.CpusetCpus, c2.CpusetMems, c2.CpuShares, c2.CpuQuota)
+	if c2.Memory == 67108864 || c2.MemorySwap == -1 || c2.CpusetCpus == "0" || c2.CpusetMems == "0" || c2.CpuShares == 100 || c2.CpuQuota == 8000 || fmt.Sprintf("%s", c2.Ulimits) == "[nofile=42:42]" {
```

Comment:
and this one I change to `c2.Ulimit != nil`, if we use `c2.Ulimits[0].Hard == 42` here, it will cause the panic since c2.Ulimits is nil, So I think we use `c2.Ulimit != nil` is ok to make sure constraints aren't saved to image.